src/NHSD.BuyingCatalogue.Documents.API/Controllers/SolutionsController.cs

```diff
@@ -11,12 +12,16 @@ namespace NHSD.BuyingCatalogue.Documents.API.Controllers
     [AllowAnonymous]
     public class SolutionsController : ControllerBase
     {
+        private readonly IStorage _storage;
+
+        public SolutionsController(IStorage storage)
+            => _storage = storage;
+
         [HttpGet]
         [Route("{id}/documents/")]
-        [ProducesResponseType(typeof(IEnumerable<string>), (int)HttpStatusCode.OK)]
-        public ActionResult GetFileNames(string id)
-        {
-            return Ok(new List<string>{"roadmap.pdf"});
-        }
+        [ProducesResponseType(typeof(IAsyncEnumerable<string>), (int)HttpStatusCode.OK)]
+        [ProducesResponseType((int)HttpStatusCode.InternalServerError)]
+        public ActionResult GetFileNames(string id) 
```

Comment:
@sam-roberts-bjss the AzureBlobStorage implementation of GetFileNames runs asynchronously and returns IAsyncEnumerable<T>. IAsyncEnumerable<T> only exposes one method, GetAsyncEnumerator(CancellationToken), which is not awaitable so it will always run synchronously.

However, both the ASP.NET Core framework ([see ObjectResultExecutor](https://github.com/aspnet/Mvc/blob/master/src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/ObjectResultExecutor.cs)) and the [IAsyncEnumerable<T>.Select implementation](https://github.com/dotnet/reactive/blob/master/Ix.NET/Source/System.Linq.Async/System/Linq/Operators/Select.cs) will process the result asynchronously.